src/main/java/com/uid2/admin/audit/AuditFactory.java

```diff
@@ -0,0 +1,37 @@
+package com.uid2.admin.audit;
+
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+
+/**
+ * AuditFactory controls the instantiation/creation of AuditMiddleware objects.
+ * Depending on the needs of the specific implementation, the AuditFactory
+ * can be implemented to always return the same AuditMiddleware object, create a new
+ * AuditMiddleware object for every class that calls getAuditMiddleware(Class), or
+ * exhibit some other behavior.
+ */
+public class AuditFactory {
+    public static AuditWriter auditWriter;
+    private static AuditMiddleware auditMiddleware;
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuditFactory.class);
+
+    /**
+     * Returns an AuditMiddleware object for the designated class to use.
+     *
+     * @param clazz the class that requires an AuditMiddleware object.
+     * @return the designated AuditMiddleware object for the passed class.
+     */
+    public static AuditMiddleware getAuditMiddleware(Class<?> clazz){
+        if(auditMiddleware == null){
+            LOGGER.fatal("setConfig not called before getAuditMiddleware in class AuditFactory");
+            throw new IllegalStateException("setConfig not called before getAuditMiddleware in class AuditFactory");
+        }
+        return auditMiddleware;
+    }
+
+    public static void setConfig(JsonObject config){
```

Comment:
why don't pass config in `getAuditMiddleware`

Reply:
I thought about doing that but initially judged it to be an awkward solution. Thanks for letting me know this is actually ok